lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -99,35 +103,36 @@ func (handler *StorageHandlersImpl) Configure(api *operations.PortLayerAPI, hand
-		_, err := vsVolumeStore.AddStore(context.TODO(), volDatastore, volStoreName)
+		_, err := vsVolumeStore.AddStore(op, volDatastore, volStoreName)
 		if err != nil {
 			log.Errorf("volume addition error %s", err)
 		}
 	}
 
-	storageVolumeLayer, err = spl.NewVolumeLookupCache(context.TODO(), vsVolumeStore)
+	h.volumeCache, err = spl.NewVolumeLookupCache(op, vsVolumeStore)
 	if err != nil {
 		log.Panicf("Cannot instantiate the Volume Lookup cache: %s", err)
 	}
 
-	api.StorageCreateImageStoreHandler = storage.CreateImageStoreHandlerFunc(handler.CreateImageStore)
-	api.StorageGetImageHandler = storage.GetImageHandlerFunc(handler.GetImage)
-	api.StorageGetImageTarHandler = storage.GetImageTarHandlerFunc(handler.GetImageTar)
-	api.StorageListImagesHandler = storage.ListImagesHandlerFunc(handler.ListImages)
-	api.StorageWriteImageHandler = storage.WriteImageHandlerFunc(handler.WriteImage)
-	api.StorageDeleteImageHandler = storage.DeleteImageHandlerFunc(handler.DeleteImage)
+	api.StorageCreateImageStoreHandler = storage.CreateImageStoreHandlerFunc(h.CreateImageStore)
+	api.StorageGetImageHandler = storage.GetImageHandlerFunc(h.GetImage)
+	api.StorageGetImageTarHandler = storage.GetImageTarHandlerFunc(h.GetImageTar)
+	api.StorageListImagesHandler = storage.ListImagesHandlerFunc(h.ListImages)
+	api.StorageWriteImageHandler = storage.WriteImageHandlerFunc(h.WriteImage)
+	api.StorageDeleteImageHandler = storage.DeleteImageHandlerFunc(h.DeleteImage)
 
-	api.StorageVolumeStoresListHandler = storage.VolumeStoresListHandlerFunc(handler.VolumeStoresList)
-	api.StorageCreateVolumeHandler = storage.CreateVolumeHandlerFunc(handler.CreateVolume)
-	api.StorageRemoveVolumeHandler = storage.RemoveVolumeHandlerFunc(handler.RemoveVolume)
-	api.StorageVolumeJoinHandler = storage.VolumeJoinHandlerFunc(handler.VolumeJoin)
-	api.StorageListVolumesHandler = storage.ListVolumesHandlerFunc(handler.VolumesList)
-	api.StorageGetVolumeHandler = storage.GetVolumeHandlerFunc(handler.GetVolume)
+	api.StorageVolumeStoresListHandler = storage.VolumeStoresListHandlerFunc(h.VolumeStoresList)
+	api.StorageCreateVolumeHandler = storage.CreateVolumeHandlerFunc(h.CreateVolume)
+	api.StorageRemoveVolumeHandler = storage.RemoveVolumeHandlerFunc(h.RemoveVolume)
+	api.StorageVolumeJoinHandler = storage.VolumeJoinHandlerFunc(h.VolumeJoin)
+	api.StorageListVolumesHandler = storage.ListVolumesHandlerFunc(h.VolumesList)
+	api.StorageGetVolumeHandler = storage.GetVolumeHandlerFunc(h.GetVolume)
 }
 
 // CreateImageStore creates a new image store
-func (handler *StorageHandlersImpl) CreateImageStore(params storage.CreateImageStoreParams) middleware.Responder {
-	url, err := storageImageLayer.CreateImageStore(context.TODO(), params.Body.Name)
+func (h *StorageHandlersImpl) CreateImageStore(params storage.CreateImageStoreParams) middleware.Responder {
+	op := trace.NewOperation(context.Background(), fmt.Sprintf("CreateImageStore(%s)", params.Body.Name))
```

Comment:
If params has a context, can we use that to chain off? That way we'd get cross layer stacks.
I'm not sure what that'd look like, but could be useful. Could also be excessive - probably not for this PR either way.